crates/core/tedge_mapper/src/c8y/tests.rs

```diff
@@ -49,6 +49,22 @@ async fn mapper_publishes_a_supported_operation_and_a_pending_operations_onto_c8
     // Start SM Mapper
     let sm_mapper = start_sm_mapper(broker.port).await;
 
+    let mut msg = messages
+        .recv()
+        .with_timeout(TEST_TIMEOUT_MS)
+        .await
+        .expect_or("No message received before timeout");
+    dbg!(&msg);
+
+    while !msg.contains("114") {
+        msg = messages
+            .recv()
+            .with_timeout(TEST_TIMEOUT_MS)
+            .await
+            .expect_or("No message received before timeout");
+        dbg!(&msg);
+    }
+
     // Expect both 114 and 500 messages has been received on `c8y/s/us`, if no msg received for the timeout the test fails.
     mqtt_tests::assert_received(
```

Comment:
This assertion is not using the `msg` read earlier. The first non-114 message would be present in that variable. Don't we want to assert what that is? Was that supposed to be the `118` message? I guess this is the reason for the test failure here: https://github.com/thin-edge/thin-edge.io/runs/5099358139?check_suite_focus=true